apps/frontend/src/app/Solver/GOSolver/BackgroundWorker.ts

```diff
@@ -25,7 +25,7 @@ async function handleEvent(e: MessageEvent<WorkerCommand>): Promise<void> {
         data.maxIterateSize
       )) {
         postMessage({ command: 'iterate', filter })
-        await Promise.resolve() // in case a `threshold` is sent over
+        await new Promise((r) => setTimeout(r)) // in case a `threshold` is sent over
```

Comment:
what does this even do?

Reply:
`Promised.resolved` returns a resolved promised, which may transparently turn into a sync computation during async-to-promise transformation. This change ensures that promise resolution happens on the next tick (and not immediately), forcing the compiler to treat that as an async computation.

Reply:
After more experiments, the difference between tasks and microtasks is the culprit. Tasks are executed in the next event loop, while microtasks are executed before the next task.

https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API/Microtask_guide#tasks_vs._microtasks